src/OpenTelemetry/Metrics/BatchMetricPoint.cs

```diff
@@ -84,6 +89,8 @@ public void Dispose()
             /// <inheritdoc/>
             public bool MoveNext()
             {
+                this.index++;
```

Comment:
We should try to reduce the calls to `metricPoints[this.index]` in this loop. I'm thinking something like...

```csharp
     public bool MoveNext()
     {
		  while (this.index < this.targetCount)
		  {
	                ref var metricPoint = ref this.metricPoints[++this.index];

	                if (metricPoint.StartTime == default)
	                {
				          continue;
        	        }

                    metricPoint.StartTime = this.start;
               	    metricPoint.EndTime = this.end;
      	            return true;
		}

        return false;
     }
```